src/components/parser.ts

```diff
@@ -159,10 +159,23 @@ function latexmkSkipped(log: string): boolean {
     return false
 }
 
+/**
+ * Convert a bibtexParser.FieldValue to a string
+ * @param field the bibtexParser.FieldValue to parse
+ */
+function fieldValueToString(field: bibtexParser.FieldValue): string {
+    if (field.kind === 'concat') {
+        return field.content.map(value => fieldValueToString(value)).reduce((acc, cur) => {return acc + ' # ' + cur})
+    } else {
+        return field.content
+    }
+}
+
 export const parser = {
     parseLatex,
     parseLatexPreamble,
     parseBibtex,
     parseLog,
-    dispose
+    dispose,
+    fieldValueToString
```

Comment:
> The `export const parser = {}` serves like a singleton object. So exposing a quite low-level `fieldValueToString` function may not look quite elegant. Let me think of a way to hide that function.

I did not pay attention to this. You are right, this looks clumsy.



> What about we make this concat-to-string a post-processing step after `parseBibtex`, so that no `concat` type would exist in the returned AST object?

I am afraid that modifying the AST object may cause more troubles that it actually solves. A `FieldValue` can be `bibtexParser.TextStringValue | bibtexParser.NumberValue | bibtexParser.AbbreviationValue | bibtexParser.ConcatValue`. So concatenating all elements inside the AST would mean to change the actual type of the field and in particular we would loose the information that one of the elements if any was an abbreviation. i would rather keep the `FieldValue` untouched. I will think of an other approach.